Database.cpp

```diff
@@ -1,8 +1,32 @@
 #include "Database.hpp"
-#include <algorithm>
 
-//void searchByLastName();
-//void searchByPersonalID();
+void Database::searchByLastName(const std::string lastName)
+{
+    std::vector<Person*>::iterator it = std::find_if(begin(data), end(data), [lastName] (Person* person) 
+            {
+                return person -> getLastName() == lastName;
+            });
+    if (it != end(data))
+    {
+        std::cout << *it << std::endl;
```

Comment:
try to keep to one style: 
`if (..)
{
aaa;
}
else 
{
return}`
 or 
`if(...)
aaaa;
else
return`

for one line after if, you don't need {}, so choose which you like :)